### ISSUE TITLE
scala.js support?

### ISSUE THREAD

**Issue by ngbinh** [andyglow/scala-jsonschema#54]
I can help with the cross-compiling but would like to check if it is something you guys are not against.

**Comment by andyglow** [comment]
that would be nice
I am not sure how useful it would be, coz to me.. 
where do we use json schema generation? - rest api, right?
how much people are doing it in `scala` to run with `node` later?

but.. anyways sounds like it may bring a lot of fun, huh? ))
wonder if adding cross-compilation with scala.js is something that can be accomplished alongside with scala-native?

**Comment by ngbinh** [comment]
scala-native gonna be hard due to https://github.com/scala-native/scala-native/issues/1629 https://github.com/scala-native/scala-native/issues/1629

**Comment by ngbinh** [comment]
The use case for me is I may need to use json-schema to define certain UI structures.

**Comment by andyglow** [comment]
so you are doing frontend (web-ui) in scala?

**Comment by ngbinh** [comment]
@andyglow yes. Been using scala.js since 2015 after years of Typescript/CoffeeScript. We now end up with about 1M LOCs frontend codebase that builds to 14 different bundles/products.

**Comment by andyglow** [comment]
@ngbinh sounds impressive! 
So yeah, if you have time and ready to invest it into Scala-jsonschema you are welcome. Besides other things it is an opportunity for me to dive into Scala.js a little )

**Comment by fdietze** [comment]
I started a PoC PR in #376. But my knowledge ends here, would be great if someone could help to finish.
